src/hotspot/share/opto/vectornode.hpp

```diff
@@ -840,6 +840,45 @@ class LoadVectorMaskedNode : public LoadVectorNode {
   Node* Ideal(PhaseGVN* phase, bool can_reshape);
 };
 
+//-------------------------------LoadVectorGatherMaskedNode---------------------------------
+// Load Vector from memory via index map under the influence of a predicate register(mask).
+class LoadVectorGatherMaskedNode : public LoadVectorNode {
+ public:
+  LoadVectorGatherMaskedNode(Node* c, Node* mem, Node* adr, const TypePtr* at, const TypeVect* vt, Node* indices, Node* mask)
+    : LoadVectorNode(c, mem, adr, at, vt) {
+    init_class_id(Class_LoadVector);
```

Comment:
class ID set at Class_LoadVector will prevent usage of [is/isa/as]_LoadVectorGatherMasked. it should be set as Class_LoadVectorGatherMasked with appropriate changes in node.hpp, but we can defer it for now till actual usages of such APIs are needed in code base.

Reply:
Yes, absolutely! I used `class_LoadVector` mainly because currently we doesn't need any usage of s/isa/as]_LoadVectorGatherMasked. This could be updated in future if needed! Thanks for you review @jatin-bhateja !